modules/wallet/history.go

```diff
@@ -0,0 +1,85 @@
+package wallet
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+var (
+	errOutOfBounds = errors.New("requesting transactions at unknown confirmation heights")
+)
+
+// ConfirmedTransactionHistory returns all of the confirmed transactions known
+// to the wallet's history.
+func (w *Wallet) ConfirmedTransactionHistory() []modules.WalletTransaction {
+	lockID := w.mu.Lock()
+	defer w.mu.Unlock(lockID)
+	return w.walletTransactions
+}
+
+// PartialTransactionHistory returns all of the confirmed transactions between
+// 'startBlock' and 'endBlock' (inclusive).
+func (w *Wallet) PartialTransactionHistory(startBlock types.BlockHeight, endBlock types.BlockHeight) ([]modules.WalletTransaction, error) {
+	lockID := w.mu.Lock()
+	defer w.mu.Unlock(lockID)
+
+	if startBlock > w.consensusSetHeight || endBlock > w.consensusSetHeight || startBlock > endBlock {
+		return nil, errOutOfBounds
+	}
+	if len(w.walletTransactions) == 0 {
+		return nil, nil
+	}
+
+	i := 0
+	for i < len(w.walletTransactions) {
+		if w.walletTransactions[i].ConfirmationHeight >= startBlock {
+			break
+		}
+		i++
+	}
+	j := i
+	for j < len(w.walletTransactions) {
+		if w.walletTransactions[j].ConfirmationHeight > endBlock {
+			break
+		}
+		j++
+	}
+	return w.walletTransactions[i:j], nil
```

Comment:
perhaps clearer:
```go
var start, end int
for i, txn := range w.walletTransactions {
    if txn.ConfirmationHeight >= startBlock {
        start = i
    }
    if txn.ConfirmationHeight > endBlock {
        end = i
        break
    }
}
return w.walletTransactions[start:end], nil
```
there may be another problem here: if `start == end`, then `[start:end] == nil`. I'm not sure if this is the desired/expected behavior.